cosinnus_message/rocket_chat.py

```diff
@@ -728,25 +730,36 @@ def users_update(self, user, request=None, force_user_update=False, update_passw
         profile = user.cosinnus_profile
         rocket_email = user_data.get('emails', [{}])[0].get('address', None)
         # rocket_mail_verified = user_data.get('emails', [{}])[0].get('verified', None)
-        rocket_username = self._get_unique_username(profile)
-        if not rocket_username:
-            return
-        if force_user_update or user_data.get('name') != rocket_username or rocket_email != profile.rocket_user_email:
+        rocket_username = user_data.get('username')
+        username_changed = rocket_username is None or self.has_username_changed(profile, rocket_username)
+        if username_changed:
+            rocket_username = self._get_unique_username(profile)
+            if not rocket_username:
+                return
```

Comment:
5. Do we `return` here and abandon the attempt to save the userprofile because `self._get_unique_username` only returns None if there was a RC error, or is there another reason? Can you add a small explanatory comment of the behaviour as a reference for the future please?

Reply:
Will add a comment.